ui/packages/catalog-ui-search/src/main/webapp/component/search-form-editor/search-form-editor.view.js

```diff
@@ -118,10 +123,50 @@ module.exports = Marionette.LayoutView.extend({
         selectionInterface: new SelectionInterface(),
         onMapLoaded: olMap => {
           this.showQueryForm(collection, id)
+          if (this.model.get('filterTemplate')) {
+            const geoFilters = (
+              this.model.get('filterTemplate').filters || [
+                this.model.get('filterTemplate'),
+              ]
+            ).filter(filter => CQLUtils.isGeoFilter(filter.type))
+            const coords = this.wktToCoords(geoFilters)
+            Common.queueExecution(() => {
+              this.map.currentView.map.zoomToExtent(coords)
+            })
+          }
         },
       })
     )
   },
+  wktToCoords(wktList) {
+    const parsedGeoCoords = _.flatten(
+      wktList.map(wkt => {
+        if (wkt.value.startsWith('POINT') && parseFloat(wkt.distance) !== 0) {
+          const center = terraformer.parse(wkt.value).coordinates
+          return new TurfCircle(center, parseFloat(wkt.distance), 64, 'meters')
+            .geometry.coordinates[0]
+        }
+        return terraformer.parse(wkt.value).coordinates
+      }),
+      true
+    )
+    return this.createCoordinatePairs(this.flatten(parsedGeoCoords))
+  },
+  flatten(arr) {
+    return arr.reduce((flat, toFlatten) => {
+      return flat.concat(
+        Array.isArray(toFlatten) ? this.flatten(toFlatten) : toFlatten
+      )
+    }, [])
+  },
+  createCoordinatePairs(arr) {
+    return arr.reduce(function(result, value, index, array) {
```

Comment:
❓ Any specific reason to use `function` instead of a fat arrow `=>` as in the previous `.reduce` instance above?